cmd/car/get.go

```diff
@@ -43,3 +54,101 @@ func GetCarBlock(c *cli.Context) error {
 	_, err = outStream.Write(blk.RawData())
 	return err
 }
+
+// GetCarDag is a command to get a dag out of a car
+func GetCarDag(c *cli.Context) error {
+	if c.Args().Len() < 3 {
+		return fmt.Errorf("usage: car get-dag [-s selector] <file.car> <root cid> <output file>")
+	}
+
+	bs, err := blockstore.OpenReadOnly(c.Args().Get(0))
+	if err != nil {
+		return err
+	}
+
+	// string to CID
+	blkCid, err := cid.Parse(c.Args().Get(1))
+	if err != nil {
+		return err
+	}
+
+	outStore, err := blockstore.OpenReadWrite(c.Args().Get(2), []cid.Cid{blkCid})
+	if err != nil {
+		return err
+	}
+
+	ls := cidlink.DefaultLinkSystem()
+	ls.StorageReadOpener = func(_ linking.LinkContext, l datamodel.Link) (io.Reader, error) {
+		if cl, ok := l.(*cidlink.Link); ok {
+			blk, err := bs.Get(cl.Cid)
+			if err != nil {
+				if err == ipfsbs.ErrNotFound {
+					if c.IsSet("strict") && c.Bool("strict") {
```

Comment:
Would it suffice to get the value straight up? Explicit or implicit value of `false` would have the same effect I think.